src/commands/mod.rs

```diff
@@ -298,29 +302,96 @@ impl DaemonControl {
+        spend_amount: bitcoin::Amount,
+        spend_feerate_vb: u64,
+        long_term_feerate_vb: u64,
+    ) -> Result<ListCoinsResult, CommandError> {
+        let mut db_conn = self.db.connection();
+
+        let candidate_coins: Vec<Coin> = db_conn
+            // TODO: for replacement transactions, may need to check `coin.spend_block.is_none()`
+            .coins(CoinType::Unspent)
+            .into_iter()
+            .filter_map(|(_, coin)| {
+                // TODO: may need to include unconfirmed coins in some cases, e.g. if insufficient confirmed funds => ancestor feerate considerations
+                if coin.block_info.is_some() {
+                    Some(coin)
+                } else {
+                    None
                 }
             })
             .collect();
-        ListCoinsResult { coins }
+        if candidate_coins.is_empty() {
+            return Err(CommandError::NoOutpoint);
+        }
+        let mut candidates: Vec<Candidate> = Vec::new();
+        for coin in &candidate_coins {
+            let coin_desc = self.derived_desc(coin);
+
+            // TODO: check what weights we need to add to `TXIN_BASE_WEIGHT` to give this candidate's weight
+            let added_weight =
+                coin_desc.witness_script().len() + (coin_desc.script_pubkey().len() * 4);
+
+            candidates.push(Candidate {
+                input_count: 1, // each candidate is a single UTXO for now
+                value: coin.amount.to_sat(),
+                weight: TXIN_BASE_WEIGHT + added_weight as u32,
+                is_segwit: true, // TODO: can we assume this or need to check for each coin?
+            });
+        }
+
+        let feerate = FeeRate::from_sat_per_vb(spend_feerate_vb as f32);
+        let long_term_feerate = FeeRate::from_sat_per_vb(long_term_feerate_vb as f32);
+
+        let target = Target {
+            value: spend_amount.to_sat(),
+            feerate,
+            min_fee: 0, // TODO: Perhaps for RBF `min_fee` will need to be passed as argument to ensure absolute fee of replacement is high enough.
+        };
+        // TODO: Choose appropriate values for drain (change output)
+        let drain = Drain {
+            weight: TXOUT_BASE_WEIGHT,
+            spend_weight: TXIN_BASE_WEIGHT,
+            value: 0, // value is ignored when using `change_policy::min_value` below
+        };
+        // Ensure any change output is not too small.
+        let change_policy = change_policy::min_value(drain, DUST_OUTPUT_SATS);
+        // TODO: is this the correct `base_weight` for `CoinSelector`?
+        let base_weight = bitcoin::Transaction {
+            input: Vec::new(),
+            output: Vec::new(),
+            lock_time: bitcoin::PackedLockTime(0),
+            version: 2,
+        }
+        .weight() as u32;
+
+        let cs = CoinSelector::new(&candidates, base_weight);
+        // TODO: use our own custom metric function.
+        let solutions = cs.branch_and_bound(Waste {
```

Comment:
Using the waste metric for coin selection could lead to huge consolidations. See for instance this comment from upstream (and the linked stackexchange discussion of the algorithm): https://github.com/bitcoindevkit/bdk/pull/924/files#diff-f1378f655c539699f8c398a3b3a11d02c939875b3e2d9fdb2a9685b9202fd70fR4-R18.

We should probably do *some* consolidation for Liana but i'm afraid using only the waste metric could lead to unreasonable selections.

Reply:
That's true, especially when the long-term feerate is higher than the spend feerate. I'll give it further thought.

As an additional safeguard, for now at least, we could let the user in the GUI edit the list of automatically selected coins when creating a new spend transaction.